llvm/utils/git/github-automation.py

```diff
@@ -298,6 +299,76 @@ def run(self) -> bool:
         return True
 
 
+class PRMergeOnBehalfInformation:
+    COMMENT_TAG = "<!--LLVM MERGE ON BEHALF INFORMATION COMMENT-->\n"
+
+    def __init__(self, token: str, repo: str, pr_number: int, author: str):
+        repo = github.Github(token).get_repo(repo)
+        self.pr = repo.get_issue(pr_number).as_pull_request()
+        self.author = author
+        self.repo = repo
+        self.token = token
+
+    def author_has_push_permission(self):
+        # https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#get-repository-permissions-for-a-user
+        response = requests.get(
```

Comment:
Are we not able to get this information through the `github` library? The documentation is usually buried, but they have had all the endpoints exposed that I've needed.

Reply:
The alternative way to do this would be to check if someone is in the llvm-committers team.  I know you can do this via the API.  Though, technically, the way it's done here is safer checking team membership.

Reply:
Yeah I got so focused on GraphQL, then REST, I forgot that the python library is a wrapper around REST.

Reply:
We can get it from python, I've done that.